### PR TITLE
drq: change view to reshape to support channels last

### PR DESCRIPTION
**Pull request by chunyuan-w**
The current implementation of `drq` does not support **channels last** input since `view` has constraints on the input size and stride (https://pytorch.org/docs/stable/generated/torch.Tensor.view.html):
```
Cannot view a tensor with shape torch.Size([1, 32, 35, 35]) and strides (39200, 1, 1120, 32) as a tensor with shape (1, 39200)!
```

Change `view` to `reshape` which returns a view if the shapes are compatible, and copies (equivalent to calling `contiguous()`) otherwise.

### CODE REVIEW

**Comment by chunyuan-w** [comment]
@xuzhao9 could you please help review this PR? Thanks!

**Comment by chunyuan-w** [comment]
cc @jgong5

**Review by jgong5 — APPROVED** [review]

**Review by xuzhao9 — APPROVED** [review]
LGTM, thanks for the improvement.
Just want to note that we should also add a comment in the code saying this is an update that doesn't exist in the upstream code: https://github.com/denisyarats/drq/blob/master/drq.py#L48

File: torchbenchmark/models/drq/drq.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def forward_conv(self, obs):
             conv = torch.relu(self.convs[i](conv))
             self.outputs['conv%s' % (i + 1)] = conv
 
-        h = conv.view(conv.size(0), -1)
+        h = conv.reshape(conv.size(0), -1)
```
Comment (xuzhao9) [inline torchbenchmark/models/drq/drq.py:46]: Can you add one-line comment mention that this is an improvement that doesn't exist in the upstream code (with link to the upstream code at https://github.com/denisyarats/drq/blob/master/drq.py#L48)?

Comment (jgong5) [reply torchbenchmark/models/drq/drq.py:46]: @xuzhao9 We can submit a PR to upstream too. Does it sound good to you?

Comment (xuzhao9) [reply torchbenchmark/models/drq/drq.py:46]: > @xuzhao9 We can submit a PR to upstream too. Does it sound good to you?

Sure, if you can submit a PR to upstream and get accepted, we don't need the comment line.

Comment (chunyuan-w) [reply torchbenchmark/models/drq/drq.py:48]: I've added one-line comment as suggested for now.
I'll submit a PR to upstream this. If it is accepted, let me submit a following-up PR later to this benchmark repo to remove the one-line comment. Is it okay?

Comment (xuzhao9) [reply torchbenchmark/models/drq/drq.py:48]: Yes, that shoulds perfect to me!

Comment (chunyuan-w) [reply torchbenchmark/models/drq/drq.py:48]: FYI, I've submitted a PR to the upstream drq repo: https://github.com/denisyarats/drq/pull/27.

**Comment by chunyuan-w** [comment]
For the CI failure, it seems unrelated with my changes:
image

And it also marks that there are 2 workflows awaiting approval:
image

@xuzhao9 could you please help on that?

**Comment by xuzhao9** [comment]
> For the CI failure, it seems unrelated with my changes: image
> 
> And it also marks that there are 2 workflows awaiting approval: image
> 
> @xuzhao9 could you please help on that?

There is a pre-existing CI failure not because of this PR. Sorry but we are still working on it.
I manually tested this PR should be safe, we can merge it despite the CI problem.

**Pull request closed without merging**